### DISCUSSION TITLE
Dynamically Changing CRON Schedule Based on System Settings

### DISCUSSION THREAD

**Discussion by jugalb-icpl** [DefectDojo/django-DefectDojo#11095] [Q&A]
Hi everyone,

I’ve successfully created a CRON job that runs every 5 minutes, controlled via an enable/disable column in the system_settings table. This works as expected. However, I now want to make the CRON schedule configurable by adding another column to the same table (for dynamic timing).

Here’s what I’ve done so far:

1. I added a new column in the system_settings table to configure the CRON interval dynamically.
2. I updated the settings_dist.py to read the CRON time from the environment:

```
#settings_dist.py
DD_AI_CRON_TIME = (int, 1)
...
AI_CRON_TIME = env('DD_AI_CRON_TIME')
...
CELERY_BEAT_SCHEDULE = {
    'genai_update_findings': {
        'task': 'dojo.tasks.update_fields_genai_cron',
        'schedule': crontab(minute='*/{}'.format(AI_CRON_TIME)),  # supposed to run at the interval specified in AI_CRON_TIME
    },
}
```

3. In the view, I tried updating AI_CRON_TIME at runtime using setattr like this:
```
if hasattr(settings, 'AI_CRON_TIME'):
    setattr(settings, 'AI_CRON_TIME', 2)  
else:
    random_num = "Not getting"
```

**Problem**
The CRON job still runs on the default interval and doesn’t pick up the updated value. The strange part is that when I print the updated AI_CRON_TIME, the change is visible on the screen, meaning the value is being updated successfully in the code. However, the CRON timing itself doesn’t seem to reflect this change.

**My Question:**
How can I correctly modify the CRON timing dynamically from the system_settings table and have it reflect immediately in the CRON job? Is there a better approach to achieve this, or do I need to restart the Celery worker after changing the timing?

Any help or insights would be greatly appreciated!

Thanks in advance!

**Comment by kiblik** [comment]
Hi @jugalb-icpl,
In Dojo, triggering/management of Cron tasks is happening in Celery-Beat. When the task is triggered, it is added to queue (redis), fetched by Celery-Worker and executed there.
So reload of the configuration of Celery-Beat is needed. But I'm not 100% sure how to trigger this kind of reload cross containers.
